From 6e8b7e921503045eb58e5b43971dfea7ae9434d1 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Thu, 24 Oct 2013 10:47:26 +0800
Subject: [PATCH 38/68] ENGR00283356 MX6SL_EVK, No image captured by driver
 after long time recording

Bitstream overflow cause the encoder error, enlarge the bitstream buffer size.

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 media/libstagefright/codecs/avc/enc/SoftAVCEncoder.cpp   | 6 ++++--
 media/libstagefright/codecs/avc/enc/src/bitstream_io.cpp | 6 ++++++
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/codecs/avc/enc/SoftAVCEncoder.cpp b/media/libstagefright/codecs/avc/enc/SoftAVCEncoder.cpp
index 1d66120..1b445b1 100755
--- a/media/libstagefright/codecs/avc/enc/SoftAVCEncoder.cpp
+++ b/media/libstagefright/codecs/avc/enc/SoftAVCEncoder.cpp
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2012 The Android Open Source Project
+ * Copyright (C) 2013 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -173,8 +174,8 @@ SoftAVCEncoder::SoftAVCEncoder(
             OMX_PTR appData,
             OMX_COMPONENTTYPE **component)
     : SimpleSoftOMXComponent(name, callbacks, appData, component),
-      mVideoWidth(176),
-      mVideoHeight(144),
+      mVideoWidth(320),
+      mVideoHeight(240),
       mVideoFrameRate(30),
       mVideoBitRate(192000),
       mVideoColorFormat(OMX_COLOR_FormatYUV420Planar),
@@ -593,6 +594,7 @@ OMX_ERRORTYPE SoftAVCEncoder::internalSetParameter(
                 mVideoHeight = def->format.video.nFrameHeight;
                 mVideoFrameRate = def->format.video.xFramerate >> 16;
                 mVideoColorFormat = def->format.video.eColorFormat;
+                ALOGI("internalSetParameter, w %d, h %d", mVideoWidth, mVideoHeight);
             } else {
                 mVideoBitRate = def->format.video.nBitrate;
             }
diff --git a/media/libstagefright/codecs/avc/enc/src/bitstream_io.cpp b/media/libstagefright/codecs/avc/enc/src/bitstream_io.cpp
index 0e3037f..f3e89ec 100644
--- a/media/libstagefright/codecs/avc/enc/src/bitstream_io.cpp
+++ b/media/libstagefright/codecs/avc/enc/src/bitstream_io.cpp
@@ -1,5 +1,6 @@
 /* ------------------------------------------------------------------
  * Copyright (C) 1998-2009 PacketVideo
+ * Copyright (C) 2013 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -15,7 +16,9 @@
  * and limitations under the License.
  * -------------------------------------------------------------------
  */
+
 #include "avcenc_lib.h"
+#include <cutils/log.h>
 
 #define WORD_SIZE 32
 
@@ -90,6 +93,9 @@ AVCEnc_Status AVCBitstreamSaveWord(AVCEncBitstream *stream)
 
     if (stream->buf_size - stream->write_pos <= (num_bits >> 3) + 2) /* 2 more bytes for possible EPBS */
     {
+        ALOGW("BitStream overflow, size %d, pos %d, bits %d",
+            stream->buf_size, stream->write_pos, num_bits);
+
         if (AVCENC_SUCCESS != AVCBitstreamUseOverrunBuffer(stream, (num_bits >> 3) + 2))
         {
             return AVCENC_BITSTREAM_BUFFER_FULL;
-- 
1.8.0

